Mage/src/main/java/mage/abilities/common/DealCombatDamageControlledTriggeredAbility.java

```diff
@@ -2,55 +2,61 @@
 
 import mage.abilities.TriggeredAbilityImpl;
 import mage.abilities.effects.Effect;
+import mage.constants.SetTargetPointer;
 import mage.constants.Zone;
+import mage.filter.common.FilterCreaturePermanent;
 import mage.game.Game;
-import mage.game.events.DamagedPlayerEvent;
+import mage.game.events.DamagedBatchEvent;
+import mage.game.events.DamagedEvent;
 import mage.game.events.GameEvent;
-import mage.game.events.GameEvent.EventType;
 import mage.game.permanent.Permanent;
 import mage.target.targetpointer.FixedTarget;
-
-import java.util.HashSet;
-import java.util.Set;
-import java.util.UUID;
+import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * @author LevelX2
  */
 public class DealCombatDamageControlledTriggeredAbility extends TriggeredAbilityImpl {
 
-    private final Set<UUID> damagedPlayerIds = new HashSet<>();
-    private final boolean setTargetPointer;
-    private final boolean onlyOpponents;
+    private final SetTargetPointer setTargetPointer;
+    private final FilterCreaturePermanent filter;
 
     public DealCombatDamageControlledTriggeredAbility(Effect effect) {
         this(Zone.BATTLEFIELD, effect);
     }
 
     public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect) {
-        this(zone, effect, false);
+        this(zone, effect, SetTargetPointer.NONE);
+    }
+
+    public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect, FilterCreaturePermanent filter) {
+        this(zone, effect, filter, SetTargetPointer.NONE, false);
     }
 
-    public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect, boolean setTargetPointer) {
+    public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect, SetTargetPointer setTargetPointer) {
         this(zone, effect, setTargetPointer, false);
     }
 
-    public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect, boolean setTargetPointer, boolean onlyOpponents) {
-        this(zone, effect, setTargetPointer, onlyOpponents, false);
+    public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect, boolean optional) {
+        this(zone, effect, SetTargetPointer.NONE, optional);
     }
 
-    public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect, boolean setTargetPointer, boolean onlyOpponents, boolean optional) {
+    public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect, SetTargetPointer setTargetPointer, boolean optional) {
+        this(zone, effect, new FilterCreaturePermanent("creatures"), setTargetPointer, optional);
+    }
+
+    public DealCombatDamageControlledTriggeredAbility(Zone zone, Effect effect, FilterCreaturePermanent filter, SetTargetPointer setTargetPointer, boolean optional) {
```

Comment:
looks like you still have a bunch of telescoping constructors. are all of them useful? could consider consolidating to have fewer constructors. and zone parameter probably only needs to be in the main constructor assuming most usages are Zone.BATTLEFIELD